test/controllers/content_items_controller_test.rb

```diff
@@ -409,6 +453,113 @@ class ContentItemsControllerTest < ActionController::TestCase
     assert_match "<li>#{I18n.t('ab_tests.find_utr_number_video_links.B')}</li>", response.body
   end
 
+  test "AB test replaces content on the log-in-file-self-assessment-tax-return page with default" do
+    content_item = content_store_has_schema_example("answer", "answer")
+    content_item["base_path"] = "/log-in-file-self-assessment-tax-return"
+    content_item["details"]["body"] = "<li>{{ab_test_sa_video_return_1}}</li>"
+
+    stub_content_store_has_item(content_item["base_path"], content_item)
+
+    request.headers["HTTP_GOVUK_ABTEST_SAVIDEORETURN1"] = nil
+
+    get :show, params: { path: path_for(content_item) }
+    assert_response :success
+    assert_no_match "{{ab_test_sa_video_return_1}}", response.body
+    assert_match "<li>#{I18n.t('ab_tests.sa_video_return_1.Z')}</li>", response.body
+  end
+
+  test "AB test replaces content on the log-in-file-self-assessment-tax-return page with variant A" do
+    content_item = content_store_has_schema_example("answer", "answer")
+    content_item["base_path"] = "/log-in-file-self-assessment-tax-return"
+    content_item["details"]["body"] = "<li>{{ab_test_sa_video_return_1}}</li>"
+
+    stub_content_store_has_item(content_item["base_path"], content_item)
+
+    request.headers["HTTP_GOVUK_ABTEST_SAVIDEORETURN1"] = "A"
```

Comment:
Turns out we've created a `with_variant` helper method which does this in a neater way - we could consider using it here

https://github.com/alphagov/govuk_ab_testing/blob/main/README.md#minitest